datahub-graphql-core/src/main/resources/app.graphql

```diff
@@ -223,6 +228,18 @@ type QueriesTabConfig {
   queriesTabResultSize: Int
 }
 
+
+"""
+Configuration for the queries tab
+"""
+type EntityProfileConfig {
+  """
+  The enum value from EntityProfileTab for which tab should be showed by default on
+  Domain profile pages. If null, rely on default sorting from React code.
+  """
+  domainDefaultTab: String
```

Comment:
We talked about doing it this way. A minor concern was whether people would expect to be able to set default tabs for any entity. If we say, domains->defaultTab in the yaml, I would also expect to be able to do dataset->defaultTab. Seeing the yaml in this PR with the included comment, would alleviate the concern though. I think the same argument could be made that they could expect to use a datasetDefaultTab, so we'd need a comment warning about the limitation either way. So that said, I'm all for a hierarchical version if it makes more sense long-term and isn't too much more work (if it is it's probably best to defer this).

We were only going to support doing it for domains at first to keep the size of the ticket smaller because we'd have to do some yaml parsing->translation to figure out what's a valid entity from the yaml and stuff like that. Just thought it might take us into the weeds. But if we're only doing domains for now we can just special case look for that.

I can't remember now if there were other concerns about the nested version off the top of my head so seems ok to me either way after second-thought!

Reply:
yeah I think the biggest concern we had with nesting Josh was in the yaml itself - we didn't want to make things unnecessary complicated and build out a whole "builder" in the yaml, but I think what we have is totally fine and we can always swap yaml stuff out later with something different - that's totally flexible.

What's not flexible is the graphql API, and if we see a future where we want different configs (no matter how we define them) for different entity profiles, I could see it making sense to nest these configs per entity type so we don't have some crazy long flat list. I think that future is resonable and I could see more configs than just default tab, so I'm leaning towards nesting only in the graphql api like john said.

still only support this situation for domains, but I think this lets us be more flexible with different types of entity profile custom configurations in the future.